[AOTI][refactor] Unify UserDefinedTritonKernel.codegen

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129378

Summary: Unify the UserDefinedTritonKernel argument codegen logic between python wrapper and cpp wrapper. This prepares for later PRs that will simplify AOTI codegen.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @chauhang

Differential Revision: [D59002226](https://our.internmc.facebook.com/intern/diff/D59002226)

desertfire committed:
[AOTI][refactor] Unify UserDefinedTritonKernel.codegen

Summary: Unify the argment codegen logic between python wrapper and cpp wrapper.

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI][refactor] Unify UserDefinedTritonKernel.codegen"


Summary: Unify the argment codegen logic between python wrapper and cpp wrapper.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames chauhang

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D59002226).

clee2000 commented:
@pytorchbot revert -m "broke inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCuda::test_triton_kernel_sympy_expr_arg_abi_compatible_cuda and a few other tests https://github.com/pytorch/pytorch/actions/runs/9680978494/job/26713689249 https://hud.pytorch.org/pytorch/pytorch/commit/52009068bc39ebc846bd37b44f5f9c5f62257778. The tests were added in https://github.com/pytorch/pytorch/pull/129301 which is before your base" -c landrace

desertfire committed:
Update on "[AOTI][refactor] Unify UserDefinedTritonKernel.codegen"


Summary: Unify the UserDefinedTritonKernel argument codegen logic between python wrapper and cpp wrapper. This prepares for later PRs that will simplify AOTI codegen.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames chauhang

Differential Revision: [D59002226](https://our.internmc.facebook.com/intern/diff/D59002226)

[ghstack-poisoned]

desertfire commented:
Close to land in https://github.com/pytorch/pytorch/pull/130796 instead
